addons/l10n_ch/models/ir_actions_report.py

```diff
@@ -66,19 +66,27 @@ def _render_qweb_pdf_prepare_streams(self, report_ref, data, res_ids=None):
                         },
                         res_ids=qr_inv_ids,
                     )
+                    # In the case of printing multiple invoices, the QR codes correctly reassigned to the appropriate invoice slot.
+                    if len(qr_res) > 1:
+                        stream_qr_pdf_combine = [value for value in qr_res.values() if value['stream'] is not None][0]
+                        for i in range(len(qr_res)-1):
+                            attachment_writer = OdooPdfFileWriter()
+                            attachment_writer.addPage(OdooPdfFileReader(stream_qr_pdf_combine['stream']).getPage(i))
+                            stream = io.BytesIO()
+                            attachment_writer.write(stream)
+                            qr_res[qr_inv_ids[i]]['stream'] = stream
 
                     for invoice_id, stream in qr_res.items():
-                        qr_pdf = OdooPdfFileReader(stream['stream'], strict=False)
-                        header_pdf = OdooPdfFileReader(header_res[invoice_id]['stream'], strict=False)
-
-                        page = header_pdf.getPage(0)
-                        page.mergePage(qr_pdf.getPage(0))
-
-                        output_pdf = OdooPdfFileWriter()
-                        output_pdf.addPage(page)
-                        new_pdf_stream = io.BytesIO()
-                        output_pdf.write(new_pdf_stream)
-                        streams_to_append[invoice_id] = {'stream': new_pdf_stream}
+                        if invoice_id:
```

Comment:
to keep diff minimal, I think you can just add `if not invoice_id: continue` at the top of the block